stl/inc/xthreads.h

```diff
@@ -33,19 +33,23 @@ struct _Stl_critical_section {
 };
 
 struct _Mtx_internal_imp_t {
-#ifdef _CRT_WINDOWS
+#if defined(_CRT_WINDOWS) || defined(UNDOCKED_WINDOWS_UCRT)
+#pragma detect_mismatch("_CRT_WINDOWS_mutex_size", "1")
 #ifdef _WIN64
     static constexpr size_t _Critical_section_size = 16;
 #else // ^^^ defined(_WIN64) / !defined(_WIN64) vvv
     static constexpr size_t _Critical_section_size = 8;
 #endif // ^^^ !defined(_WIN64) ^^^
-#else // ^^^ defined(_CRT_WINDOWS) / !defined(_CRT_WINDOWS) vvv
+#else // ^^^ defined(_CRT_WINDOWS) || defined(UNDOCKED_WINDOWS_UCRT) / !defined(_CRT_WINDOWS) ||
+      // defined(UNDOCKED_WINDOWS_UCRT) vvv

```

Comment:
The complement here is improperly De Morgan-ed. Given how ridiculously long it is, I think we should just change this to "Windows private STL" and "public STL".